server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -87,9 +92,46 @@ public Map<String, NodeUsageStatsForThreadPools> simulatedNodeUsageStatsForThrea
                 adjustedNodeUsageStatsForThreadPools.put(entry.getKey(), entry.getValue());
             }
         }
+
+        // Add `NodeUsageStatsForThreadPools` for any nodes not present in the original `NodeUsageStatsForThreadPools` map.
+        addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(adjustedNodeUsageStatsForThreadPools);
+
         return Collections.unmodifiableMap(adjustedNodeUsageStatsForThreadPools);
     }
 
+    private void addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(
+        Map<String, NodeUsageStatsForThreadPools> nodeUsageStatsForThreadPools
+    ) {
+        // Assume the new node has the same size thread pool as the largest existing node
+        final OptionalInt largestWriteThreadPool = originalNodeUsageStatsForThreadPools.values()
+            .stream()
+            .map(NodeUsageStatsForThreadPools::threadPoolUsageStatsMap)
+            .map(m -> m.get(ThreadPool.Names.WRITE))
+            .mapToInt(NodeUsageStatsForThreadPools.ThreadPoolUsageStats::totalThreadPoolThreads)
+            .max();
```

Comment:
Assuming the max thread pool size is probably "optimistic", we could also be pessimistic and assume the minimum pool size